tests/WP_Ultimo/Managers/Event_Manager_Test.php

```
@@ -0,0 +1,151 @@
+		$manager->do_event('test_fire', ['sample' => 'data']);
+
+		$this->assertTrue($generic_fired, 'wu_event action should have fired.');
+		$this->assertTrue($specific_fired, 'wu_event_test_fire action should have fired.');
+	}
+
+	/**
+	 * Test save_event with a fully valid payload creates an event record.
+	 */
+	public function test_save_event_with_valid_payload(): void {
+
+		$manager = $this->get_manager_instance();
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$manager'. (undefined)

(UnusedLocalVariable)